modules/build/src/main/scala/scala/build/Build.scala

```diff
@@ -212,41 +213,47 @@ object Build {
             scope,
             logger,
             buildClient,
-            bloopServer
+            bloopServer,
+            isTest
           )
 
           value(res)
         }
 
       val mainBuild = value(doBuildScope(mainOptions, mainSources, Scope.Main))
 
-      val testBuild = value {
-        mainBuild match {
-          case s: Build.Successful =>
-            val extraTestOptions = BuildOptions(
-              classPathOptions = ClassPathOptions(
-                extraClassPath = Seq(s.output)
-              )
-            )
-            val testOptions0 = extraTestOptions.orElse(testOptions)
-            doBuildScope(testOptions0, testSources, Scope.Test)
-          case _ =>
-            Right(Build.Cancelled(
-              inputs,
-              sharedOptions,
-              Scope.Test,
-              "Parent build failed or cancelled"
-            ))
+      val testBuildOpt =
+        if (isTest) {
+          val testBuild = value {
+            mainBuild match {
+              case s: Build.Successful =>
+                val extraTestOptions = BuildOptions(
+                  classPathOptions = ClassPathOptions(
+                    extraClassPath = Seq(s.output)
+                  )
+                )
+                val testOptions0 = extraTestOptions.orElse(testOptions)
+                doBuildScope(testOptions0, testSources, Scope.Test)
+              case _ =>
+                Right(Build.Cancelled(
+                  inputs,
+                  sharedOptions,
+                  Scope.Test,
+                  "Parent build failed or cancelled"
+                ))
+            }
+          }
+          Some(testBuild)
         }
-      }
+        else None
 
       doPostProcess(mainBuild, inputs0, Scope.Main)
-      doPostProcess(testBuild, inputs0, Scope.Test)
+      if (testBuildOpt.nonEmpty) doPostProcess(testBuildOpt.get, inputs0, Scope.Test)
```

Comment:
More idiomatic way of doing that (without a `.get` in particular):
```suggestion
      for (testBuild <- testBuildOpt)
        doPostProcess(testBuild, inputs0, Scope.Test)
```